app/src/main/java/com/itsaky/androidide/utils/ColorUtil.java

```diff
@@ -0,0 +1,56 @@
+package com.itsaky.androidide.utils;
+
+import android.graphics.Color;
+import android.util.Log;
+
+import androidx.core.graphics.ColorUtils;
+
+import com.itsaky.androidide.app.StudioApp;
+
+import java.lang.reflect.Field;
+
+public class ColorUtil {
+
+    public static boolean isDark(int color) {
+        return ColorUtils.calculateLuminance(color) < 0.25;
+    }
+
+    public static Object getAndroidColor(String clazz, String colorName) {
+        Field declaratedField;
+        try {
+            declaratedField = Class.forName(clazz).getField(colorName);
+            declaratedField.setAccessible(true);
+            return declaratedField.get(null);
+        } catch (NoSuchFieldException | ClassNotFoundException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static int getColor(String colorValue) {
```

Comment:
Why not use [`CommonParseUtils.parseColor`](https://github.com/itsaky/AndroidIDE/blob/235145a92f611920d4bc0d14d499560a59b458ee/xml-inflater/src/main/java/com/itsaky/inflater/util/CommonParseUtils.java#L248) here? Every call to this can be replaced by the `CommonParseUtils` implementation.

Reply:
Okay